sql/core/src/test/scala/org/apache/spark/sql/DataFrameWriterV2Suite.scala

```diff
@@ -841,20 +841,24 @@ class DataFrameWriterV2Suite extends QueryTest with SharedSparkSession with Befo
   }
 
   test("SPARK-51281: create/replace file source tables") {
+    def checkResults(df: DataFrame): Unit = {
+      checkAnswer(df, spark.range(10).toDF())
+    }
```

Comment:
How about declare `df = spark.range(10)` here ?

Reply:
what do you mean?

Reply:
How about
```
    val ds = spark.range(10)
    val df = ds.toDF()
    Seq(true, false).foreach { ignorePath =>
      withSQLConf(SQLConf.LEGACY_DF_WRITER_V2_IGNORE_PATH_OPTION.key -> ignorePath.toString) {
        withTable("t1", "t2") {
          ds.writeTo("t1").using("json").create()
          checkAnswer(spark.table("t1"), df)

          withTempPath { p =>
            val path = p.getCanonicalPath
            ds.writeTo("t2").using("json").option("path", path).create()
            checkAnswer(spark.table("t2"), df)
            if (ignorePath) {
              assert(!p.exists())
            } else {
              checkAnswer(spark.read.json(path), df)
            }
          }
        }
      }
    }
```

Reply:
let's not over-engineering. `spark.range(10)` is already very short.